db schema and some explanation

cootook opened the pull request:
Take a look at my db schema, please.

All explanations are in README.md

cootook committed:
db schema and some explanation

Pierre-Thibault commented on README.md:
```diff
@@ -1,4 +1,23 @@
+Each budget has at least one dedicated account to store money. 
+There are different types of budget (regular, goal, etc.)
+
+Account keeps monet and can belong to only budget.
```
Money

cootook replied:
Thanks for you review!
I think my explanation was not clear enough.

```In transaction, from_account_id and to_account_id should be nullable.``` that's right. They are nullable actually:

![image](https://github.com/DevMentorDave/DeepPocketsHomework/assets/65569159/1f223903-abc7-417a-aa55-d633148b1326)

![image](https://github.com/DevMentorDave/DeepPocketsHomework/assets/65569159/19c46d25-8121-41ea-aee6-cee79846e507)

This diagram editor allows to set `not null`, so all fields are nullable by default.

```I don't think a link between transaction and scheduled_transation is needed.```
I think that it is needed because the app should be able to distinguish scheduled transaction for the purpose of checking if it was done . If I have a planned transaction and the app is going to do it, first the app should be able to check if it was done already. 

```A scheduled_transaction can between two different accounts or not. Should it be linked to a budget? ``` It is a good point. Now I think, that scheduled transaction should be linked to account only not to budget. An account is linked to particular budget already. Otherwise it is going to mess different layers of abstraction. In my mind money and transactions are the base level of abstraction. A budget is a hire layer on top of money. 
There should be a rule that a scheduled transaction belongs to an account money go from, if it is earning and there is null in `from_account_id` than a scheduled transaction belongs to an account money go to.

```I don't think we need a schedule_type. This type is only in the application logic.``` Somehow the app should know which logic to apply to a particular scheduled transaction. So I think schedule_type table  has a finite  set of types (weekly, biweekly, monthly, daily, one-time). For each of that types there should be specific functionality in the app. Maybe for that table id, name and description is enough. 

p.s.: I am not able to push any commit
```
remote: error: GH013: Repository rule violations found for refs/heads/db-shema-cootok.
remote: Review all repository rules at http://github.com/DevMentorDave/DeepPocketsHomework/rules?ref=refs%2Fheads%2Fdb-shema-cootok
remote:
remote: - Changes must be made through a pull request.
remote:
To https://github.com/DevMentorDave/DeepPocketsHomework.git
 ! [remote rejected] db-shema-cootok -> db-shema-cootok (push declined due to repository rule violations)
error: failed to push some refs to 'https://github.com/DevMentorDave/DeepPocketsHomework.git'
```

Pierre-Thibault reviewed (changes requested):
In transaction, from_account_id and to_account_id should be nullable. I don't think a link between transaction and scheduled_transation is needed. A scheduled_transaction can between two different accounts or not. Should it be linked to a budget? I don't think we need a schedule_type. This type is only in the application logic.

cootook committed:
fix typo

cootook committed:
change explanation

while answering to review the whole schema was changed
the diagram is going to be changed later to fit explanation

cootook commented:
@Pierre-Thibault @DevMentorDave 
My explanation was changed but schema was not yet. So it does not fit the explanation. 
If you have any notes please share. The diagram is going to be redone later.
Thanks.

cootook committed:
change schema

cootook committed:
delete out of date sign

cootook commented:
Updated once again.
